Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -161,6 +161,71 @@ public final class SwiftLanguageServer: ToolchainLanguageServer {
     }
   }
 
+  /// Update lexical and syntactic tokens for the given `snapshot`.
+  func updateLexicalAndSyntacticTokens(
+    response: SKDResponseDictionary,
+    for snapshot: DocumentSnapshot
+  ) {
+    let uri = snapshot.document.uri
```

Comment:
This should be guarded by `queue`, either through a `queue.async` or by requiring that it’s called on `queue. Based on how `updateLexicalAndSyntacticTokens` is called right now, I think the latter is what we want. Could you add a comment and assertion for that?